File: marinetes/monorepo-legacy/packages/database/src/models/Schedule.ts
import type {
  Schedule,
  Service,
  ScheduleStatus,
  SchedulePayment,
  UserAddress,
} from '@marinetes/types/model';
import {
  Entity,
  Column,
  BaseEntity,
  PrimaryGeneratedColumn,
  CreateDateColumn,
  UpdateDateColumn,
  ManyToMany,
  JoinTable,
} from 'typeorm';

@Entity('schedules')
export class ScheduleRepository extends BaseEntity implements Schedule {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column('varchar')
  status: ScheduleStatus;

  @Column('datetime')
  date: string;

  @Column('datetime', { nullable: true })
  end_date: string;

  @Column('decimal')
  price: number;

  @Column('int')
  estimated_time_in_hours: number;

  @Column('boolean')
  verified: boolean;

  @Column('boolean')
  confirmed: boolean;

  @Column('boolean')
  going_to_local: boolean;

  @Column('char')
  user_id: string;

  @Column('char')
  diarist_id: string;

  @Column('char', { nullable: true })
  second_diarist_id: string;

  @Column('varchar')
  verification_code: string;

  @Column('varchar')
  confirmation_code: string;

  @Column('json')
  address: UserAddress;

  @Column('json')
  payment: SchedulePayment;

  @ManyToMany('services')
  @JoinTable({
    name: 'schedules_services_services',
    joinColumn: {
      name: 'schedule_id',
      referencedColumnName: 'id',
    },
    inverseJoinColumn: {
      name: 'service_id',
      referencedColumnName: 'id',
    },
  })
  services: Service[];

  @CreateDateColumn()
  created_at: string;

  @UpdateDateColumn()
  updated_at: string;
}

File: marinetes/user-mobile/app/src/main/kotlin/com/marinetes/network/serializations/marinetes/user/api/Schedule.kt
package com.marinetes.network.serializations.marinetes.user.api

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class Schedule(
  val id: String,
  val status: String,
  val date: String,
  @SerialName("end_date")
  val endDate: String? = null,
  val price: Int,
  @SerialName("estimated_time_in_hours")
  val estimatedTimeInHours: Int,
  val verified: Boolean,
  val confirmed: Boolean,
  @SerialName("going_to_local")
  val goingToLocal: Boolean,
  @SerialName("user_id")
  val userId: String,
  @SerialName("diarist_id")
  val diaristId: String,
  val diarist: Diarist?,
  @SerialName("second_diarist_id")
  val secondDiaristId: String? = null,
  @SerialName("verification_code")
  val verificationCode: String,
  @SerialName("confirmation_code")
  val confirmationCode: String,
  val address: UserAddress,
  val payment: Payment? = null,
  val services: List<Service>,
  @SerialName("created_at")
  val createdAt: String,
  @SerialName("updated_at")
  val updatedAt: String,
)